kernel/Kconfig

```diff
@@ -343,6 +343,15 @@ config THREAD_MAX_NAME_LEN
 	  Thread names get stored in the k_thread struct. Indicate the max
 	  name length, including the terminating NULL byte. Reduce this value
 	  to conserve memory.
+
+config THREAD_RUNTIME_STATS
+	bool "Thread runtime statistics"
+	help
+	  Gather thread runtime statistics.
```

Comment:
@dcpleung @nashif how does this _really_ differ from TRACING CPU_STATS?
I think there should be one way to extract thread run-time measurements (and then also obtain statistics and duty-cycling) and it should be used for Tracing, Benchmarking, etc. What do you think`?

Reply:
> how does this _really_ differ from TRACING CPU_STATS?

- TRACING_CPU_STATS is per cpu, this is per thread.
- TRACING_CPU_STATS is actually "mis-using" the tracing feature for capturing timing information, it does not trace anything.
- With TRACING_CPU_STATS enabled, you can't do tracing at the same time, so it is mutually exclusive feature, the above i independent, so you can do tracing and capture runtime stats per thread.
- Once we have this in and satisfies all TRACING_CPU_STATS features, we should consider dropping TRACING_CPU_STATS.

Reply:
OK, thanks for the clarification.
Yeap, I also had the understanding that CPU_STATS is capturing timing information (for calculating duty-cycling).

So, do we agree that THREAD_RUNTIME_STATS should be the only Kconfig switch for enabling timestamp capturing in ARCHitecture and Kernel functions?
Then THREAD_RUNTIME_STATS should be used for both bench-marking and tracing?

Reply:
> Then THREAD_RUNTIME_STATS should be used for both bench-marking and tracing?

those are 2 different things, not sure what you mean here.